### PR TITLE
feat: push image with retry logic

### PR DESCRIPTION
**Pull request by zhujian7**
**What this PR does / why we need it**:

Add your description

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #

Reference to #

**Special notes for your reviewer**:

/cc @supereagle @hyy0322 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: build/resolver/image/entrypoint.sh [hunk 1]
```diff
@@ -70,8 +70,25 @@ END
...
 fi
 
-pull() {
-    docker pull $IMAGE
+# push image with retry logic
+push() {
+    i=1
+    while [ $i -le 3 ]
+    do
+        echo "attempt number: $i"
```
Comment (supereagle) [inline build/resolver/image/entrypoint.sh]: Need to output this only if push failed.

File: build/resolver/image/entrypoint.sh [hunk 2]
```diff
@@ -70,8 +70,25 @@ END
...
+        i=$(( i + 1 ))
+        docker push $IMAGE
+        status=$?
+        if [ $status -eq 0 ]
+        then
+          echo "pushed $IMAGE successfully."
+          break
+        else
+          echo "pushed $IMAGE failed."
+          [ $i -le 3 ] && sleep 10
```
Comment (supereagle) [inline build/resolver/image/entrypoint.sh]:
```suggestion
          [ $i -le 3 ] && sleep 10
          echo "number of push retries: $i"
```

Comment (zhujian7) [reply build/resolver/image/entrypoint.sh]: PTAL

**Pull request merged**